pkg/audit/verifier.go

```diff
@@ -369,9 +374,24 @@ func (verifier *Verifier) Reverify(ctx context.Context, stripe *Stripe) (report
 				}
 
 				if errs2.IsRPC(err, codes.NotFound) {
+
+					// Get the original segment pointer in the metainfo
+					oldPtr, err := verifier.metainfo.Get(ctx, pending.Path)
+					remoteSegment := oldPtr.GetRemote()
+					if err != nil || remoteSegment == nil || remoteSegment.RootPieceId != pending.PieceID {
+						ch <- result{nodeID: piece.NodeId, status: skipped}
+						verifier.log.Debug("Reverify: audit source deleted before reverification", zap.Stringer("Node ID", piece.NodeId), zap.Error(err))
+						return
+					}
+					//remove failed audit pieces from the pointer so as to only penalize once for failed audits
+					err = verifier.RemoveFailedPieces(ctx, pending.Path, oldPtr, storj.NodeIDList{pending.NodeID})
```

Comment:
There is another place where reverification fails the audit -- line 416 (hashes mismatch).